src/opm/parser/eclipse/EclipseState/Schedule/TimeMap.cpp

```diff
@@ -184,15 +188,20 @@ namespace {
         }
     }
 
-    void TimeMap::addFromTSTEPKeyword(const DeckKeyword &TSTEPKeyword) {
+    void TimeMap::addFromTSTEPKeyword(const DeckKeyword &TSTEPKeyword, bool TSTEPUnitIsHours) {
         if (TSTEPKeyword.name() != "TSTEP")
             throw std::invalid_argument("Method requires TSTEP keyword input.");
         {
             const auto &item = TSTEPKeyword.getRecord(0).getItem(0);
 
             for (size_t itemIndex = 0; itemIndex < item.size(); itemIndex++) {
-                const double days = item.get<double>(itemIndex);
-                const int64_t seconds = static_cast<int64_t>(days * 24 * 60 * 60);
```

Comment:
This should just be fixed by using:
```
double seconds = item.getSIDouble(itemIndex);
```
then it should correct - for any unit system.

Reply:
> This should just be fixed by using:
> 
> ```c++
> double seconds = item.getSIDouble(itemIndex);
> ```

Yes, good idea.  That's the correct solution&mdash;possibly `static_cast<>` to `std::int64_t` to comply with requirements of [`TimeMap::addTStep()`](https://github.com/OPM/opm-common/blob/d40177dacb9e9891e16d3c6bedc6f9d5b48a3cbe/src/opm/parser/eclipse/EclipseState/Schedule/TimeMap.cpp#L137-L139)

Reply:
Cool!